elasticsearch/src/http/transport.rs

```diff
@@ -363,7 +380,14 @@ impl Transport {
         B: Body,
         Q: Serialize + ?Sized,
     {
-        let connection = self.conn_pool.next();
+        if self.conn_pool.reseedable() {
+            self.conn_pool.reseeding();
+            // NodesInfo::new(&self, NodesInfoParts::None)
+            //     .send()
+            //     .await
+            //     .expect("Could not retrieve nodes for refresh");
```

Comment:
This is technically a recursive call, which requires returning a `futures::future::BoxFuture`, still playing around with making that work without breaking the current `send` implementation.

Reply:
My current understanding (concurrency is hard so it may be wrong :D), is that if two threads get the `reseedable() == true` it'll kick off two serial reseeds as the reseed routine acquires a write lock (I don't predict this to be a likely event but definitely needs to be accounted for. I thought this trade off would be ok accounting for complexity of adding a subroutine for triggering reseeds). The problem I see is reseeding would block the current `ConnectionPool` requests until the Write lock is dropped. If you think kicking off multiple serial requests may be problematic I am more than willing to change the implementation to a background operation. :)

Reply:
I envisage the implementation doing something like the following:

1. An API request is initiated
2. The connection pool indicates if it is reseedable and needs reseeding
3. If reseeding is needed, a thread-safe signal is set to indicate reseeding is in progress
4. A reseed request is started, either in _synchronous_ flow of the request, or on a separate thread. A separate thread is probably preferable as the request that initiates reseeding is not then impacted by the reseeding process. It does make the implementation more complex though.
5. Upon receiving the reseed response, a write lock is acquired to update the connection pool. Failure in the reseeding process, should be logged (I think we can add a TODO for this for now and skip, as the client does not currently have logging in place, but should have in future).
6. A thread-safe signal is set to indicate reseeding is not in progress